client/src/pages/Signup.js

```diff
@@ -30,6 +36,10 @@ class SignUpPage extends Component {
     this.setState({ password: event.target.value });
   };
 
+  handleConfirmPasswordChange = event => {
+    this.setState({ confirmPassword: event.target.value });
+  };
+
   validate = () => {
     let emailError = "";
     let passwordError = "";
```

Comment:
Instead of just a validate, I was thinking to have a validation for each form item (this allows the errors to go away once their validations are met). 

  emailValidate = () => {
    let emailError = "";
    var re = /[^@]+@[^.]+\..+/;
    var test = re.test(this.state.email);
    if (!test === true) {
      emailError = "Invalid email";
    }

    if (emailError) {
      this.setState({ emailError });
      return false;
    }
    this.setState({ emailError: undefined });
    return true;
  };

  nameValidate = () => {
    let nameError = "";

    if (this.state.name.length < 1) {
      nameError = "Please enter your name";
    }

    if (nameError) {
      this.setState({ nameError });
      return false;
    }
    this.setState({ nameError: undefined });
    return true;
  };

  passwordValidate = () => {
    let passwordError = "";

    if (this.state.password.length < 6) {
      passwordError = "Password is too short";
    }

    if (passwordError) {
      this.setState({ passwordError });
      return false;
    }
    this.setState({ passwordError: undefined });
    return true;
  };